### ISSUE TITLE
help for DateTime hard to access

### ISSUE THREAD

**Issue by JeffBezanson** [JuliaLang/julia#11438]
There is loads of great help for Dates but I hit some speed bumps trying to get to it. `DateTime` needed to be quoted to work with `help`:

```
julia> help(DateTime)
DataType   : Base.Dates.DateTime
  supertype: Base.Dates.TimeType
  fields   : [:instant]

julia> help("DateTime")
Dates.DateTime

   "DateTime" wraps a "UTInstant{Millisecond}" and interprets it
   according to the proleptic Gregorian calendar.
...
```

Not sure why that's happening.
Also some functions are grouped together in a way that help/ReST doesn't support:

```
julia> help(Dates.year)
Dates.year(dt::TimeType) -> Int64
month(dt::TimeType) -> Int64
week(dt::TimeType) -> Int64
day(dt::TimeType) -> Int64
hour(dt::TimeType) -> Int64
minute(dt::TimeType) -> Int64
second(dt::TimeType) -> Int64
millisecond(dt::TimeType) -> Int64

   Return the field part of a Date or DateTime as an "Int64".

julia> help(Dates.month)
month (generic function with 5 methods)
```

**Comment by fcard** [comment]
The first problem happens with a lot of types.
```julia
julia> help(Tridiagonal)
DataType   : Base.LinAlg.Tridiagonal{T}
  supertype: AbstractArray{T, 2}
  fields   :  [:d1, :d, :du, :du2]

julia> help("Tridiagonal")
Base.Tridiagonal(d1, d, du)
    ...
    ...

julia> help(RoundingMode)
DataType   : Base.Rounding.RoundingMode{T}
  ...

julia> help("RoundingMode")
Base.RoundingMode...

julia> help(Libc.TmStruct)
DataType   : Base.Libc.TmStruct
  ...

julia> help("Libc.TmStruct")
Libc.TmStruct...
```

In fact, the only one I got to work was `Array`.
I think the help database has trouble extracting types from modules outside Base, in general.

**Comment by fcard** [comment]
I fixed half of this with #11538.

**Comment by KristofferC** [comment]
Can be closed I think?:

```
help?> Dates.year
  year(dt::TimeType) -> Int64

  The year of a Date or DateTime as an Int64.
```

```
help?> DateTime
search: DateTime broadcast_getindex broadcast_setindex! InvalidStateException

  DateTime

  DateTime wraps a UTInstant{Millisecond} and interprets it according to the
  proleptic Gregorian calendar.

  DateTime(y, [m, d, h, mi, s, ms]) -> DateTime

  Construct a DateTime type by parts. Arguments must be convertible to Int64.

  DateTime(periods::Period...) -> DateTime

  Construct a DateTime type by Period type parts. Arguments may be in any
  order. DateTime parts not provided will default to the value of
  Dates.default(period).

  DateTime(dt::Date) -> DateTime

  etc etc lots of nice stuff
```
